front/lib/api/files/text_extraction.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  isTextExtractionSupportedContentType,
+  pagePrefixesPerMimeType,
+  TextExtraction,
+} from "@dust-tt/types";
+import { buffer } from "stream/consumers";
+
+import config from "@app/lib/api/config";
+import type { Authenticator } from "@app/lib/auth";
+import type { FileResource } from "@app/lib/resources/file_resource";
+import logger from "@app/logger/logger";
+
+export const extractTextFromFile = async (
+  auth: Authenticator,
+  file: FileResource
+): Promise<string> => {
+  try {
+    if (!isTextExtractionSupportedContentType(file.contentType)) {
+      throw new Error("unsupported_content_type");
+    }
+
+    const readStream = file.getReadStream({
+      auth,
+      version: "original",
+    });
+
+    // Load file in memory.
+    const arrayBuffer = await buffer(readStream);
+
+    // Extract text from the file.
+    const extractionRes = await new TextExtraction(
+      config.getTextExtractionUrl()
+    ).fromBuffer(arrayBuffer, file.contentType);
+
+    if (extractionRes.isErr()) {
+      // We must throw here, stream does not support Result type.
+      throw extractionRes.error;
+    }
+
+    const pages = extractionRes.value;
+
+    const prefix = pagePrefixesPerMimeType[file.contentType];
+
+    let content: string = "";
+    for (const page of pages) {
+      const pageText = prefix
+        ? `${prefix}: ${page.pageNumber}/${pages.length}\n${page.content}\n\n`
+        : page.content;
+      content += pageText;
```

Comment:
We use to do this operation with a stream to avoid loading the whole file in memory? Why do we move away from the stream approach?

Reply:
We were loading everything in memory upfront even in the previous version.
However, as discussed IRL, will do another update to bring full streaming support to text extraction.